### PR TITLE
[NavigationDrawer] Performance improvements and increased support for customizing initial drawer percentage height

### PR DESCRIPTION
**Pull request by yarneo**
This is the first part of multiple improvements to get the Navigation Drawer to the state that it can receive an arbitrary percentage from 0%-100% of how much of the drawer to present on the screen when displayed (currently it is only at 50% and is uncustomizable).

This PR allows us to set the drawer an arbitrary number without it breaking, with the caveat that it doesnt yet support percentages that are 100% or close to 100%. Once that is supported we will be able to support 100% drawer presented for VO, solving bug #4899 

This also has performance improvements as before when scrolling the drawer up with the trackingScrollView being set, it would call `viewWillLayoutSubviews` because it would think for an instance that it is not in full screen anymore, before reverting back to the correct state.

**Testing:**
Currently to test this addition, I have ran all the examples that provide all the needed use cases of usage for the drawer. They have been run in both portrait and landscape, and by dragging the drawer from the start position, to full screen, to edge of content, and back up again. This process has been run on an iPhone SE with iOS 9.4, an iPhone X with iOS 11.4, an iPhone 6 Plus with iOS 10, and an iPad running iOS 11.4.

I can't see any edge cases with the current support we allow with this component, with the addition that this is an alpha component.

### CODE REVIEW

**Comment by jverkoey** [comment]
Please add a description of how you're testing this behavior and any edge cases to be aware of to the PR description.

**Comment by yarneo** [comment]
Done

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m [hunk 1]
```diff
@@ -215,6 +217,7 @@ - (void)updateContentOffsetForPerformantScrolling:(CGFloat)contentYOffset {
       // Update the drawer's scrollView's offset to be static so the content will scroll instead.
       CGRect scrollViewBounds = self.scrollView.bounds;
       scrollViewBounds.origin.y = drawerOffset;
+      normalizedYContentOffset = drawerOffset;
```
Comment (jverkoey) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:226]: What does normalized mean in this context?

**Review by jverkoey — APPROVED** [review]
LGTM but I can't say that I fully understand what's happening here. This may benefit from some additional documentation explaining the meaning of the terms and the mappings (though if that already exists let me know!).

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m (hunk 1 above)
Comment (yarneo) [reply components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:226]: I've updated the code with an elaborate explanation of what this variable entails.
